src/pocketmine/plugin/PluginManager.php

```diff
@@ -752,7 +752,16 @@ public function registerEvents(Listener $listener, Plugin $plugin) : void{
 				$ignoreCancelled = isset($tags["ignoreCancelled"]) && strtolower($tags["ignoreCancelled"]) === "true";
 
 				$parameters = $method->getParameters();
-				if(count($parameters) === 1 and $parameters[0]->getClass() instanceof \ReflectionClass and is_subclass_of($parameters[0]->getClass()->getName(), Event::class)){
+				try{
+					$isHandler = count($parameters) === 1 && $parameters[0]->getClass() instanceof \ReflectionClass && is_subclass_of($parameters[0]->getClass()->getName(), Event::class);
+				}catch(\ReflectionException $e){
+					if(isset($tags["softDepend"])){
+						continue;
+					}
+
+					throw new \ClassNotFoundException($e->getMessage(), 0, $e);
```

Comment:
I would say it's better to just rethrow `$e` here (as per the original behaviour). `ClassNotFoundException` is not expected to be thrown by this function and is not used anywhere else (it remains solely for backwards compatibility and will be removed in future).